src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
Oh, right, that's true.

You could also enforce slightly more rigorous splits, i.e. at least 80%/20%, yielding better accuracy at the cost of taking longer to build the tree.